operator/pkg/client/factory.go

```diff
@@ -172,7 +171,7 @@ func (s *SingletonClients) clientCacheKeyForMapping(mapping *meta.RESTMapping) s
 }
 
 func (s *SingletonClients) DynamicResourceInterface(obj *unstructured.Unstructured) (dynamic.ResourceInterface, error) {
-	mapping, err := getResourceMapping(obj, s.discoveryShortcutExpander)
+	mapping, err := getResourceMapping(obj, s.discoveryShortcutExpander, false)
```

Comment:
why is this set to false?

Reply:
This is set as SetApplier currently always resets the mapper already in `adjustNs`, however I understand this is only a workaround so I will correct it to true